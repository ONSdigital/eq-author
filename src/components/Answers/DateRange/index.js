import React from "react";
import PropTypes from "prop-types";
import CustomPropTypes from "custom-prop-types";
import Date from "../Date";

const DateRange = ({ answer, ...otherProps }) =>
  answer.childAnswers.map(answer => (
    <Date key={answer.id} answer={answer} {...otherProps} />
  ));

DateRange.propTypes = {
  answer: CustomPropTypes.answer.isRequired,
  onUpdate: PropTypes.func.isRequired
};

export default DateRange;
