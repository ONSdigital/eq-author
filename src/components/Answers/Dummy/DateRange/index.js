import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import DummyTextInput from "../TextInput";
import { colors } from "constants/theme";
import chevronIcon from "components/Accordion/chevron.svg";

const GUTTER = "0.625em";

const Field = styled.div`
  display: inline-block;
  width: calc(${props => props.colWidth} - ${GUTTER});
  margin-left: ${GUTTER};

  &:first-of-type {
    margin-left: 0;
  }
`;

Field.propTypes = {
  colWidth: PropTypes.string.isRequired
};

const Input = styled(DummyTextInput)`
  width: auto;
  border-radius: 3px;
  padding: 0.875em 1em;
  border-color: #bfbfbf;
  color: #bfbfbf;
`;

const Select = styled(Input)`
  &::after {
    content: url(${chevronIcon});
    position: absolute;
    right: 0.75em;
    top: 0.75em;
  }
`;

const Label = styled.p`
  color: ${colors.text};
  font-size: 0.875em;
  font-weight: bold;
  margin-top: 0;
  margin-bottom: 0.75rem;
`;

const Fieldset = styled.div`
  margin-bottom: 1.125em;

  &:last-of-type {
    margin-bottom: 0;
  }
`;

const Legend = styled.p`
  font-size: 1.125em;
  font-weight: bold;
  color: ${colors.text};
  margin-top: 0;
  margin-bottom: 0.875rem;
`;

export const DatePicker = () => (
  <div>
    <Field colWidth="25%">
      <Label>Day</Label>
      <Input>DD</Input>
    </Field>
    <Field colWidth="50%">
      <Label>Month</Label>
      <Select>Select month</Select>
    </Field>
    <Field colWidth="25%">
      <Label>Year</Label>
      <Input>YYYY</Input>
    </Field>
  </div>
);

const StyledDateRange = styled.div.attrs({
  role: "presentation"
})`
  cursor: default;
  user-select: none;
`;

const DateRange = () => (
  <StyledDateRange>
    <Fieldset>
      <Legend>Period from</Legend>
      <DatePicker />
    </Fieldset>
    <Fieldset>
      <Legend>Period to</Legend>
      <DatePicker />
    </Fieldset>
  </StyledDateRange>
);

export default DateRange;
