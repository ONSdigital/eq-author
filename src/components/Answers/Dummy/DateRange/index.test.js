import React from "react";
import { shallow } from "enzyme";
import DateRange, { DatePicker } from "./index";

describe("Dummy/DateRange", () => {
  it("should render", () => {
    const wrapper = shallow(<DateRange />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render dummy DatePicker", () => {
    const wrapper = shallow(<DatePicker />);
    expect(wrapper).toMatchSnapshot();
  });
});
