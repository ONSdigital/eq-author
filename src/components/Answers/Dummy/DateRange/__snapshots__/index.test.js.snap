// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dummy/DateRange should render 1`] = `
<DateRange__StyledDateRange>
  <DateRange__Fieldset>
    <DateRange__Legend>
      Period from
    </DateRange__Legend>
    <DatePicker />
  </DateRange__Fieldset>
  <DateRange__Fieldset>
    <DateRange__Legend>
      Period to
    </DateRange__Legend>
    <DatePicker />
  </DateRange__Fieldset>
</DateRange__StyledDateRange>
`;

exports[`Dummy/DateRange should render dummy DatePicker 1`] = `
<DateRange__Flex>
  <DateRange__Field>
    <DateRange__Label>
      Day
    </DateRange__Label>
    <DateRange__Input
      placeholder="DD"
    />
  </DateRange__Field>
  <DateRange__SelectField>
    <DateRange__Label>
      Month
    </DateRange__Label>
    <DateRange__Select
      placeholder="Select month"
    />
  </DateRange__SelectField>
  <DateRange__Field>
    <DateRange__Label>
      Year
    </DateRange__Label>
    <DateRange__Input
      placeholder="YYYY"
    />
  </DateRange__Field>
</DateRange__Flex>
`;
